### DISCUSSION TITLE
Problem with jsPsych.randomization.factorial in conjunction with  image-keyboard-response plugin

### DISCUSSION THREAD

**Discussion by kennysmithed** [jspsych/jsPsych#1377] [Q&A]
I am having an issue in 6.2.0 for code that was working OK in 6.1.0, involving jsPsych.randomization.factorial in combination with image-keyboard-response trials. Code like this worked under 6.1.0:

```js
var factors = {
    stimulus: ['Dog1.jpg', 'Dog2.jpg', 'Dog3.jpg'],
    prompt: ['<p>prompt 1</p>', '<p>prompt 2</p>', '<p>prompt 3</p>']
};

var factorial_values = jsPsych.randomization.factorial(factors,1);

console.log(factorial_values)

var trial = {
    type: 'image-keyboard-response',
    stimulus: jsPsych.timelineVariable('stimulus'),
    prompt:jsPsych.timelineVariable('prompt'),
    on_start: function(trial) {console.log (trial.stimulus,trial.prompt)}
};

var trials_with_variables = {
    timeline: [trial],
    timeline_variables: factorial_values
};
```
However, under 6.2.0 it produces an unexpected behaviour where the images are only shown on some trials. If I swap the image-keyboard-response plugin for html-keyboard-response then it behaves as expected (i.e. the image name is displayed as the stimulus on all trials). Looking at the plugins it seems that the way images are handled has changed, is there some interaction with the factorial function that's causing the problem or am I doing something else wrong? Any help much appreciated! Kenny

**Comment by jodeleeuw** [comment]
Hi @kennysmithed,

We did change the way that the `image-keyboard-response` plugin presents images, to address a few bugs. This created a new bug that we didn't catch in testing. We are tracking it here #1305, and it will be fixed with `v6.3`. You can revert the plugin to the old method by adding `render_on_canvas: false` as an option in the trial. Or you can preload the images in `jsPsych.init`, which should also fix the bug (and improve timing in the experiment).

**Reply by kennysmithed** [reply]
Thanks for the very quick response! I am still getting the same behaviour with render_on_canvas:false in the trial actually, but it goes away if I tell it to preload the images in jsPsych.init.
